packages/transaction-controller/src/TransactionController.ts

```diff
@@ -438,14 +451,34 @@ export class TransactionController extends BaseController<
       transactions: [],
     };
     this.initialize();
-    const provider = getProvider();
+    this.provider = provider;
     this.getNetworkState = getNetworkState;
     this.ethQuery = new EthQuery(provider);
     this.registry = new MethodRegistry({ provider });
+    this.nonceTracker = new NonceTracker({
+      provider,
+      blockTracker,
+      getPendingTransactions: (address) => {
+        return formatTxForNonceTracker(
+          this.state.transactions.filter(
+            ({ status, transaction: { from } }) =>
+              status === TransactionStatus.submitted &&
+              (address ? from.toLowerCase() === address.toLowerCase() : true),
+          ),
+        );
+      },
+      getConfirmedTransactions: () => {
```

Comment:
This accepts an address as well; we'd want to ensure we're considering only the confirmed transactions for the given account.

Reply:
not sure why I missed this 🤔

Reply:
addressed here: https://github.com/MetaMask/core/pull/1147/commits/2138d654c19b87a7e9baf2221fad057ed3434975